specs/HubProtocol.md

```diff
@@ -20,13 +20,17 @@ This document describes two encodings of the SignalR protocol: [JSON](http://www
 
 In the SignalR protocol, the following types of messages can be sent:
 
-* `Negotiation` Message - Sent by the client to negotiate the message format.
-* `Invocation` Message - Indicates a request to invoke a particular method (the Target) with provided Arguments on the remote endpoint.
-* `StreamInvocation` Message - Indicates a request to invoke a streaming method (the Target) with provided Arguments on the remote endpoint.
-* `StreamItem` Message - Indicates individual items of streamed response data from a previous Invocation message.
-* `Completion` Message - Indicates a previous Invocation or StreamInvocation has completed. Contains an error if the invocation concluded with an error or the result of a non-streaming method invocation. The result will be absent for `void` methods. In case of streaming invocations no further `StreamItem` messages will be received
-* `CancelInvocation` Message - Sent by the client to cancel a streaming invocation on the server.
-* `Ping` Message - Sent by either party to check if the connection is active.
+| Message Name          | Sender         | Description                                                                                                                    |
+| ------------------    | -------------- | ------------------------------------------------------------------------------------------------------------------------------ |
+| `NegotiationRequest`  | Caller         | Sent by the client to negotiate the message format.                                                                            |
+| `NegotiationResponse` | Callee         | Sent by the server as an acknowledgment of the previous `NegotiationRequest` message. Contains an error if negotiation failed. |
+| `Close`               | Callee         | Sent by the server when a connection is closed. Contains an error if the connection was closed because of an error.            |
```

Comment:
This should be `Callee, Caller` since it could be either party that decides it has to close the connection.

Hey @davidfowl , this mean we kinda have our `/abort` endpoint from SignalR 2.x back, it's just in the SignalR layer, not the underlying transports. The client can send `Close` and the server can simply tear down the connection immediately, allowing cooperative clients to clean up Long Polling resources without having to wait for the poll interval to elapse.